### DISCUSSION TITLE
convert_xpath_to_css does not support long xpath

### DISCUSSION THREAD

**Discussion by STARKILLER-1** [seleniumbase/SeleniumBase#2732] [Q&A] [answered]
How do i click a button via Diver() and  xpath: `"//button[contains(@class, 'btn-main') and .//span[contains(@class, 'hide-mobile') and contains(text(), 'Добавить в корзину')]]"`
convert_xpath_to_css does not work

**Comment by mdmintz** [comment, marked as answer]
Not every XPath has a matching CSS Selector. There isn't even a `:contains("TEXT")` selector in CSS. SeleniumBase has a special `TAG:contains("TEXT")` selector, which gets converted to XPath before use, but you need to make a JavaScript call, which requires a normal CSS Selector (and that doesn't allow any kind of `:contains("TEXT")` selector).

This means that you should either provide a CSS Selector directly, or an XPath Selector that can be converted into a natural CSS Selector. Otherwise you'll get errors, the `uc_click(selector)` won't work, or the `uc_click(selector)` won't click with stealth.

**Reply by STARKILLER-1** [reply]
I figured it out, there was a fake element protecting button(had same attributes). I delete it via `driver.execute_script("var element = document.querySelector('.product-page__order'); if (element) element.remove();")` and then css selector works `driver.click('button[class="btn-main"]')` 

Sorry for million of stupid questions =)
